src/plugins/intel_cpu/src/emitters/snippets/riscv64/jit_memory_emitters.hpp

```diff
@@ -0,0 +1,108 @@
+// Copyright (C) 2018-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <cstddef>
+#include <memory>
+#include <nodes/kernels/riscv64/cpu_isa_traits.hpp>
+#include <nodes/kernels/riscv64/jit_generator.hpp>
+#include <vector>
+
+#include "emitters/plugin/riscv64/jit_emitter.hpp"
+#include "openvino/core/type/element_type.hpp"
+#include "snippets/lowered/expression.hpp"
+
+namespace ov::intel_cpu::riscv64 {
+
+class jit_memory_emitter : public jit_emitter {
+public:
+    jit_memory_emitter(ov::intel_cpu::riscv64::jit_generator_t* h,
+                       ov::intel_cpu::riscv64::cpu_isa_t isa,
+                       const ov::snippets::lowered::ExpressionPtr& expr,
+                       emitter_in_out_map in_out_type);
+
+    size_t aux_gprs_count() const override;
+
+    void emit_code_impl(const std::vector<size_t>& in_idxs,
+                        const std::vector<size_t>& out_idxs,
+                        const std::vector<size_t>& pool_vec_idxs,
+                        const std::vector<size_t>& pool_gpr_idxs) const override;
+
+    std::vector<size_t> get_available_aux_gprs() const;
+
+protected:
+    static size_t get_parent_buffer_cluster_id(const ov::snippets::lowered::ExpressionPtr& expr);
+    static size_t get_consumer_buffer_cluster_id(const ov::snippets::lowered::ExpressionPtr& expr);
```

Comment:
Let's move these functions to some utils and reuse it for all platforms memory emitters